observability/sanitize.go

```
@@ -0,0 +1,415 @@
+		return walkValue(value.Elem(), key, depth+1, seen)
+	}
+
+	if value.CanInterface() {
+		switch typed := value.Interface().(type) {
+		case string:
+			return Sanitize(typed)
+		case error:
+			return Sanitize(safeError(typed))
+		case fmt.Stringer:
+			return Sanitize(safeString(typed))
+		case json.Number:
```

golangci-lint (2.12.2)

[error] SA4020: unreachable case clause: fmt.Stringer will always match before encoding/json.Number

(staticcheck)